sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -835,5 +922,82 @@ public async ValueTask DisposeAsync()
             await CloseAsync().ConfigureAwait(false);
             GC.SuppressFinalize(this);
         }
+
+        /// <summary>
+        /// Updates the concurrency for the processor. This method can be used to dynamically change the concurrency of a running processor.
+        /// </summary>
+        /// <param name="maxConcurrentCalls">The new max concurrent calls value. This will be reflected in the <see cref="ServiceBusProcessor.MaxConcurrentCalls"/>
+        /// property.</param>
+        public void UpdateConcurrency(int maxConcurrentCalls)
+        {
+            Argument.AssertAtLeast(maxConcurrentCalls, 1, nameof(maxConcurrentCalls));
+            lock (_maxConcurrencySyncLock)
```

Comment:
do we still need to lock?

Reply:
Yes, to guard against concurrent calls to Update causing non-atomic updates.